.github/workflows/ci.yml

```diff
@@ -22,22 +22,22 @@ jobs:
   lint:
     runs-on: ubuntu-latest
     needs: changes
+    if: needs.changes.outputs.theirs == 'true'
     steps:
-      - name: Checkout
-        if: needs.changes.outputs.theirs == 'true'
-        uses: actions/checkout@v2
-      - name: Lint Markdown
-        if: needs.changes.outputs.theirs == 'true'
-        uses: docker://avtodev/markdown-lint:v1
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
         with:
-          args: 'posts/${{ github.event.pull_request.user.login }}/**/*.md'
+          node-version: 17
+      - run: npm ci
+      - run: npx --no-install markdownlint **/*.md'
+        working-directory: posts/${{ github.event.pull_request.user.login }}
   scope:
     runs-on: ubuntu-latest
     needs: changes
+    if: needs.changes.outputs.others == 'true'
```

Comment:
This is not correct, the reason it was within the step was so that build checks would pass for other type of PRs that don't necessarily affect content related to posts. Now they get skipped but they are a requirement on the branch protection rules.